src/index.js

```diff
@@ -46,14 +51,78 @@ function Id (id, privKey, pubKey) {
   }
 }
 
-// generation
+//unwrap the private key protobuf stream 
+function unmarshal (key) {
+  var dpb = messages.PrivateKey.decode(key)
+  return dpb
+}
+
+//create a public key protobuf to be base64 string stored in config
+function marshal (data, type) {
+  if(type == 'Public'){
+    var epb = messages.PublicKey.encode({
+      Type: 0,
+      Data: data
+    })
+  }
+
+  if(type == 'Private'){
+    var epb = messages.PrivateKey.encode({
+      Type: 0,
+      Data: data
+    })
+  }
+
+  return epb
+}
 
+// generation
 exports.create = function () {
-  var pair = keypair()
+  //generate keys
+  var pair = forge.rsa.generateKeyPair({bits:2048, e: 0x10001})
+
+  //Create Public Key
```

Comment:
The creation of the pubkey comes first than the privOne? Bit counter intuitive. Would make sense to make both part of different functions so that the code can be used below on `createFromPrivKey`?